src/Streamly/Prelude.hs

```diff
@@ -155,6 +156,20 @@ foldrM step acc m = go (toStream m)
             yield a (Just x) = step a (go x)
         in (runStream m1) Nothing stop yield
 
+-- | Scan left. A strict left fold which accumulates the result of its reduction steps inside a stream, from left.
+{-# INLINE scan #-}
+scan :: Streaming t => (x -> a -> x) -> x -> (x -> b) -> t m a -> t m b
+scan step begin done m = cons (done begin) $ fromStream $ go (toStream m) begin
+    where
+    go m1 !acc = Stream $ \_ stp yld ->
+        let
+          stop = stp
```

Comment:
Need 4 space indent here as well as in the nested let clause. Also, I prefer to write multiline let clauses like this (not hard and fast rule though):

```
let stop = stp
    yield a Nothing = ...
```

The first line of code comes right after the `let` keyword.

Sorry for being a pain in the neck about style and indent :-)

Reply:
Oh! I actually write `let` like this too. I copied your style in the `foldl` function.